src/hooks/useCampaignManager.js

```diff
@@ -13,7 +13,11 @@ export const useCampaignManager = () => {
 
   const getCampaigns = async (userId, postId) => {
     const signer = await provider?.getSigner();
-    const CampaignManager = new Contract(addresses.CampaignManager, CampaignManagerJson, signer);
+    const CampaignManager = new Contract(
```

Comment:
Can you define `CampaignManager` just once in this file?

Reply:
i cant use await outside of that function